src/main/java/io/confluent/connect/hdfs/avro/AvroRecordWriterProvider.java

```diff
@@ -85,6 +85,12 @@ public void write(SinkRecord record) {
           }
         } catch (IOException e) {
           throw new DataException(e);
+        } finally {
+          try {
+            close();
+          } catch (Throwable t) {
+            log.error("Could not close Record Writer", t);
```

Comment:
You may want to verify in a scratch area that the DataException will still be thrown. I do recall that an exception on the finally block will override the one caught before the finally block, but I don't recall what happens if the finally block catches it.

Reply:
I've verified that the original exception gets thrown correctly.